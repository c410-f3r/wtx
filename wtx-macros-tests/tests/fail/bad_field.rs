#[wtx_macros::pkg(data_format(json), id(FooId))]
mod pkg {
  #[derive(Debug)]
  #[pkg::req_data]
  pub struct Req(#[pkg::bar] i32);

  #[pkg::res_data]
  struct Res;
}

#[wtx_macros::pkg(data_format(json), id(FooId))]
mod pkg {
  #[derive(Debug)]
  #[pkg::req_data]
  pub struct Req(#[pkg::field(foo = "1111")] i32);

  #[pkg::res_data]
  struct Res;
}

fn main() {
}
