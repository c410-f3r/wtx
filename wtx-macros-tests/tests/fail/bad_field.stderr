error: Field attributes must be annotated as `#[pkg::field(name = "SomeName")]`
 --> tests/fail/bad_field.rs:5:20
  |
5 |   pub struct Req(#[pkg::bar] i32);
  |                    ^^^

error: expected `name`
  --> tests/fail/bad_field.rs:15:31
   |
15 |   pub struct Req(#[pkg::field(foo = "1111")] i32);
   |                               ^^^

error: aborting due to 2 previous errors

